src/main/java/net/dries007/tfc/world/vein/VeinType.java

```diff
@@ -59,7 +59,7 @@ public VeinType(ResourceLocation id, JsonObject json)
             throw new JsonParseException("Vertical Size must be > 0.");
         }
         density = JSONUtils.getAsInt(json, "density", 20) / 100f;
-        if (density <= 0 || density > 100)
+        if (density <= 0 || density > 1)
```

Comment:
Just wanted to point out this. I've found out that it seems to be an overlook, so I've fixed it in this branch.

The problem here is that you aren't checking for the user input but rather the value after it being divided by 100. It seems you wanted it to be 0-1 (and the user input between 0-100), hence the change.

Reply:
Yup. Everything there needs a rewrite anyway at some point. *sigh*.